components/region_cache_memory_engine/src/range_manager.rs

```diff
@@ -298,30 +298,68 @@ impl RangeManager {
         vec![]
     }
 
-    // return whether the range can be already removed
-    pub(crate) fn evict_range(&mut self, evict_range: &CacheRange) -> bool {
-        let Some(range_key) = self
-            .ranges
-            .keys()
-            .find(|&r| r.contains_range(evict_range))
-            .cloned()
-        else {
+    /// Return ranges that can be deleted now (not ongoing snapshot).
+    // There are two cases based on the relationship between `evict_range` and
+    // cached ranges:
+    // 1. `evict_range` is contained(including equals) by a cached range (only one
+    //    due to non-overlapping in cached ranges)
+    // 2. `evict_range` is overlapped with (including contain but not be contained)
+    //    one or more cached ranges
+    //
+    // For 1, if the `evict_range` is a proper subset of the cached_range, we will
+    // split the cached_range so that only the `evict_range` part will be evicted
+    // and deleted.
```

Comment:
Should we adjust tags and ranges of the partially evicted range? Because region split creates new region id and key ranges.

Reply:
tag is just used for print

Reply:
May be clarify and document that tag is not to be used for identification purposes, only for debugging.